### PR TITLE
fix a potential race condition

### PR DESCRIPTION
**Pull request by oroulet**

### CODE REVIEW

File: asyncua/common/connection.py [hunk 1]
```diff
@@ -264,6 +269,9 @@ def _check_sym_header(self, security_hdr):
             self.revolve_tokens()
             return
 
+        if self._renewal_started and security_hdr.TokenId == self.security_token.TokenId + 1:
```
Comment (Randelung) [inline asyncua/common/connection.py:272]: This just disables the check, yes? Potential security issue. Not a fix (the underlying race condition still exists), but a workaround.
I can think of at least the wrap-around case where the right-hand check won't hold up.

File: asyncua/common/connection.py [hunk 2]
```diff
@@ -277,6 +285,11 @@ def _check_sym_header(self, security_hdr):
             return
 
         expected_tokens = [self.security_token.TokenId, self.next_security_token.TokenId]
+
+        if self._renewal_started:
+            extra_token = self.security_token.TokenId + 1
```
Comment (Randelung) [inline asyncua/common/connection.py:290]: Is the next token always `n+1`? The [spec](https://reference.opcfoundation.org/v104/Core/docs/Part4/5.5.2/) doesn't specify any relation between old and new, potentially making a random new token possible, making prediction impossible. Is it defined somewhere else?

Comment (oroulet) [reply asyncua/common/connection.py:290]: it make it accept temporarily the next token. But yes this is not nice. I mainly made that code to test if this solves the issue. So far I got only one report and it is negative....

Comment (oroulet) [reply asyncua/common/connection.py:290]: OK. you say tthat it fixes it... I need to have a look at all these merge request...

**Pull request closed without merging**